File: src/test/java/testingil/webinar/tddapi/s11/refactor_test/NavigatorUnitTests.java
package testingil.webinar.tddapi.s11.refactor_test;

import static org.hamcrest.CoreMatchers.is;
import static org.mockito.Mockito.RETURNS_DEEP_STUBS;
import static org.mockito.Mockito.mock;

import org.junit.jupiter.api.Test;
import testingil.webinar.tddapi.s11.Distance;
import testingil.webinar.tddapi.s11.Location;
import testingil.webinar.tddapi.s11.Navigator;
import testingil.webinar.tddapi.s11.DistanceProvider;

import static org.hamcrest.MatcherAssert.assertThat;

class NavigatorUnitTests {

	@Test
	void when_created_distance_is_always_zero() {
		Location loc = new Location("London");
		DistanceProvider mockDistService = 
				mock(DistanceProvider.class, RETURNS_DEEP_STUBS);
		Navigator nav = new Navigator(loc, mockDistService);
		Distance distance = nav.getDistanceFromDestination();
		assertThat(distance.inKm(), is(0));
	}
	
//	@Test
//	@DisplayName("2500 miles = 4000 kms")
//	void distance_is_calculated_in_km() {
//		Location startingPoint = new Location("New York City");
//		Navigator nav = new Navigator(startingPoint);
//		Location dest = new Location("Los Angeles");
//		nav.setDestination(dest);
//		Distance distance = nav.getDistanceFromDestination();
//		assertThat(distance.inKm(), is(4000));
//	}

}

File: Python/s14_refactor_test/navigator_unit_tests.py
import unittest
from unittest.mock import patch

from hamcrest import assert_that, equal_to

from s14_refactor_test.distance import Distance
from s14_refactor_test.location import Location
from s14_refactor_test.navigator import Navigator


class NavigatorUnitTests(unittest.TestCase):
    def setUp(self) :
        self.distProvider = patch('s14_refactor_test.distance_adapter.DistanceAdapter').start()

    def test_when_created_distance_is_always_zero(self):
        self.set_distance_to(0)
        nav = Navigator(Location("London"), self.distProvider)
        distance = nav.get_distance_from_destination()
        assert_that(distance.inKm(), equal_to(0))

    def test_distance_is_calculated_in_km(self):
        """2500 miles = 4000 kms"""
        self.set_distance_to(2500)
        startingPoint = Location("New York City")
        nav = Navigator(startingPoint, self.distProvider)
        nav.set_destination(Location("Los Angeles"))
        distance = nav.get_distance_from_destination()
        assert_that(distance.inKm(), equal_to(4000))

    def set_distance_to(self, miles):
        self.distProvider.get_distance.return_value = Distance(miles)


if __name__ == '__main__':
    unittest.main()
